app/service/rest_svc.py

```diff
@@ -187,6 +188,17 @@ def __init__(self):
         operation = (await self.get_service('data_svc').locate('operations', match=dict(id=link.operation)))[0]
         return await operation.apply(link)
 
+    async def task_agent_with_ability(self, paw=None, ability_id=None):
+        abilities = await self.get_service('data_svc').locate('abilities', match=dict(ability_id=ability_id))
+        if abilities:
+            for op in await self.get_service('data_svc').locate('operations', match=dict(state='running')):
```

Comment:
all of these if's and for's can be consolidated with a little thought muscle.

Reply:
I can, but then we will just have errors for index/list access in the event of malformed data (which is on a lot of the functions inside `rest_svc`

Reply:
```python
    async def task_agent_with_ability(self, paw=None, ability_id=None):
        abilities = (await self.get_service('data_svc').locate('abilities', match=dict(ability_id=ability_id)))[0]
        for op in await self.get_service('data_svc').locate('operations', match=dict(state='running')):
            agent = [agent for agent in await op.active_agents() if agent.paw == paw]
            if agent and await agent[0].capabilities(ability_set=abilities):
                return await op.build_and_apply_custom_link(agent=agent[0],
                                                            ability=await agent[0].capabilities(
                                                                ability_set=abilities[0])[0],
                                                            operation_id=op.id)
```

Reply:
I've got one other approach I am trying...